Switched to different color matching system; made the sensor look for the color that lines up with the one π/2 away

TheHarmonicRealm opened the pull request:
Switched to built-in way of checking colors; made it so that the placeholder color tests correspond to the right ones so it actually has the right idea about it.

Commit:
before pull

Commit:
Merge branch 'master' of https://github.com/Spartronics4915/2020-InfiniteRecharge

Commit:
renamed to be clearer what that does

Commit:
moved stopExtendMotor to be with the other motor-stopping methods

Commit:
minor ordering changes

Commit:
I pulled

Commit:
fixed a problem in the commented-out part

Commit:
before changing things to make this work for testing

Commit:
before switching to a different color-determining system

Commit:
Merge branch 'master' of https://github.com/Spartronics4915/2020-InfiniteRecharge

Commit:
first attempt at using the built-in way; much is copied from https://github.com/REVrobotics/Color-Sensor-v3-Examples/blob/master/Java/Color%20Match/src/main/java/frc/robot/Robot.java

Commit:
new method for testing better

Commit:
this is what I am using to test it

Commit:
Merge branch 'master' of https://github.com/Spartronics4915/2020-InfiniteRecharge

Commit:
Justice said to commit now

Commit:
before pulling

Commit:
after pulling and  fixing merge conflicts

Commit:
made it so that the colors aren't added every times; added comments

Commit:
fixed sneaky merge conflict

Commit:
it should warn me about those merge conflicts before I can add...

Commit:
fixed error

Commit:
adjusted to new placeholders

Commit:
clearer diagram

Commit:
made minor method for Martin

omentic commented on src/main/java/com/spartronics4915/frc2020/Constants.java:
```diff
@@ -182,4 +196,34 @@
         public static final Pose2d kCameraOffset = new Pose2d();
         public static final double kMeasurementCovariance = 1;
     }
+<<<<<<< HEAD
```
Unresolved merge conflict

omentic commented on src/main/java/com/spartronics4915/frc2020/Constants.java:
```diff
@@ -83,6 +87,15 @@
 
     public static final class PanelRotator
     {
+<<<<<<< HEAD
```
Unresolved merge conflict

omentic commented on src/main/java/com/spartronics4915/frc2020/Constants.java:
```diff
@@ -91,6 +104,7 @@
 
         public static final double kRaiseSpeed = 0.5;
         public static final double kLowerSpeed = -0.5;
+>>>>>>> 6994853066370995031bb4e51dab95c757325bfd
```
Cruft!

omentic commented on src/main/java/com/spartronics4915/frc2020/Constants.java:
```diff
@@ -55,7 +58,8 @@
         public static final int kOpticalFlagId = 2; // A2 // Optical Flag for Zeroing
```
These values need to be located inside of the PanelHandler subclass.

omentic commented on src/main/java/com/spartronics4915/frc2020/Constants.java:
```diff
@@ -55,7 +58,8 @@
                                                                   // is inverted
-        public static final int kIntakeSensorId = -1; // 
+        //TODO: sounds misplaced…needs a better name?
+        public static final int kIntakeSensorId = -1;
```
This sensor has a port assigned to it - check the [2020 Control Map spreadsheet.](https://docs.google.com/spreadsheets/d/14O3Dj1Bl6tqiHFrCJ7wyVRDOqS5EZ11p4q9j2rYS-3Q/edit)

If you want a different name, I'd suggest `kProximitySensorId`.

omentic commented on src/main/java/com/spartronics4915/frc2020/Constants.java:
```diff
@@ -68,9 +72,9 @@
-        public static final int kAngleAdjusterMasterId = 0; // PWM0
-        public static final int kAngleAdjusterFollowerId = 1; // PWM1
+        public static final int kFlywheelFollowerId = 0; // Solid brass
+        public static final int kAngleAdjusterMasterId = 0;
```
Add comments again, it's helpful to be able to distinguish these from CAN devices.

omentic commented on src/main/java/com/spartronics4915/frc2020/Constants.java:
```diff
@@ -68,9 +72,9 @@
-        public static final int kFlywheelFollowerId = -1; // Solid brass
-        public static final int kAngleAdjusterMasterId = 0; // PWM0
-        public static final int kAngleAdjusterFollowerId = 1; // PWM1
+        public static final int kFlywheelFollowerId = 0; // Solid brass
```
Revert this to -1. 

-1 is the convention for invalid ids, which this is - it's only here as a placeholder if another motor does get added.

omentic commented on src/main/java/com/spartronics4915/frc2020/Constants.java:
```diff
@@ -83,6 +87,15 @@
     public static final class PanelRotator
     {
+<<<<<<< HEAD
+        public static final int kBeamSensorUpID = 0;
```
These are the wrong ids and have the wrong capitalization.

omentic commented on src/main/java/com/spartronics4915/frc2020/commands/PanelRotatorCommands.java:
```diff
@@ -125,17 +131,17 @@ public void end(boolean interrupted)
-        private int eighths;
-        private String lastColor;
-        private String currentColor;
+        public int eighths;
```
These should not be public!

omentic commented on src/main/java/com/spartronics4915/frc2020/subsystems/PanelRotator.java:
```diff
@@ -119,56 +145,65 @@ public String getTargetColor()
+        {
+            sensedColor = "Red";
+        }
+        else if (match.color == kGreenTarget)
```
`==` are not the correct way to compare Strings.

omentic reviewed (changes requested):
You need to go through and deal with the unresolved merge conflicts, as well as implement other feedback.

Commit:
pulled and fixed stuff that Justice did not like

Commit:
that's the ID

Commit:
moved optical flag stuff from Commands.Indexer to Commands.PanelRotator

Commit:
oops I destroyed Camden's thing

Commit:
limit switch, names

Commit:
fixed the problems

Commit:
getColorForDashboard now also compares the strings in the right way

Commit:
I think that that is the right id?

Commit:
more IDs

omentic committed:
Rename and correct some port ids

omentic committed:
Use proper constants, fix variable visibility, and clean up structure

omentic committed:
Update Constants.java

omentic committed:
Move color targets to Constants.java

omentic committed:
Update PanelRotator to pull from Constants.java

omentic committed:
Reference correct method

omentic committed:
Fix Gradle build

omentic committed:
Allow multiple assignments of motors

omentic committed:
Use switch statement over if/else block and remove duplicate method

Switch statements use `.equals()` on Strings.
https://docs.oracle.com/javase/7/docs/technotes/guides/language/strings-switch.html

omentic committed:
Revert change to switch statement

Colors are not Strings.

omentic committed:
Fix  illegitimate String comparison

omentic committed:
Remove unnessary chassis check

???

omentic committed:
Revert that

omentic committed:
Revert again? (I'm bad at GitHub web editor)

omentic committed:
Third attempt at a fix

omentic committed:
Fourth attempt - this should work.

omentic committed:
Merge branch 'master' into master
